Build errors when cross-compiling for Linux on Windows with the UE Marketplace plugin version

When cross-compiling for Linux on Windows with the UE Marketplace plugin version the following build errors occur:

```
UATHelper: Packaging (Linux): Missing precompiled manifest for 'Sentry', '<EngineRoot>\Engine\Plugins\Marketplace\Sentry\Intermediate\Build\Linux\UnrealGame\Development\Sentry\Sentry.precompiled'. This module was most likely not flagged for being included in a precompiled build - set 'PrecompileForTargets = PrecompileTargetsType.Any;' in Sentry.build.cs to override. If part of a plugin, also check if its 'Type' is correct.
UATHelper: Packaging (Linux): Took 7,06s to run dotnet.exe, ExitCode=6
UATHelper: Packaging (Linux): UnrealBuildTool failed. See log for more details. (C:\Users\<Username>\AppData\Roaming\Unreal Engine\AutomationTool\Logs\D+unreal-editors+UE_5.3\UBT-MarketplaceTest-Linux-Development.txt)
UATHelper: Packaging (Linux): AutomationTool executed for 0h 0m 9s
UATHelper: Packaging (Linux): AutomationTool exiting with ExitCode=6 (6)
UATHelper: Packaging (Linux): BUILD FAILED
PackagingResults: Error: Unknown Error
```

Presumably, the required pre-compiled Linux binaries that should be distributed along with the plugin are getting obfuscated.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -237,7 +237,7 @@ jobs:
         id: pre-compile-linux
         run: |
           docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux -NoHostPlatform
-          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry ls -al /dist
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al /dist
 
       - name: Upload Linux intermediates
         uses: actions/upload-artifact@v3.1.1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -242,7 +242,7 @@ jobs:
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/Intermediate/Build/Linux          
+          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/Intermediate/Build/Linux/*
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -242,7 +242,7 @@ jobs:
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/dist/HostProject/Plugins/Sentry/Intermediate/Build/Linux          
+          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/dist/Intermediate/Build/Linux          
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -281,18 +281,19 @@ jobs:
         with:
           name: ${{ github.sha }}
 
-      - name: Extract packages
-        run: unzip sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip -d sentry-unreal-*-engine${{ matrix.unreal }}-marketplace
+      - name: Extract package
+        run:
+          unzip sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip -d sentry-unreal-engine${{ matrix.unreal }}-marketplace
 
       - name: Download Linux intermediates
         uses: actions/download-artifact@v3
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: sentry-unreal-*-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate
+          path: sentry-unreal-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate
 
       - name: Prepare Sentry packages for release
         shell: pwsh
-        run: zip -r -1 -v sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip ./*
+        run: zip -r -1 -v sentry-unreal-engine${{ matrix.unreal }}-marketplace.zip ./sentry-unreal-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate/*
 
       - name: Re-upload package
         uses: actions/upload-artifact@v3.1.1
@@ -301,4 +302,4 @@ jobs:
           name: ${{ github.sha }}
           if-no-files-found: error
           path: |
-            sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip
+            sentry-unreal-engine${{ matrix.unreal }}-marketplace.zip

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -237,6 +237,9 @@ jobs:
         id: pre-compile-linux
         run: |
           docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al /Intermediate
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al /Intermediate/Build
 
       - name: Upload Linux intermediates
         uses: actions/upload-artifact@v3.1.1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -236,7 +236,7 @@ jobs:
       - name: Pre-compile Linux intermediates
         id: pre-compile-linux
         run: |
-          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux -NoHostPlatform
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux
           docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al /dist
 
       - name: Upload Linux intermediates

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -242,7 +242,7 @@ jobs:
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/dist/Intermediate/Build/Linux          
+          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/dist/HostProject/Intermediate/Build/Linux          
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27
@@ -262,3 +262,44 @@ jobs:
             checkout/${{ matrix.app }}/Saved/Stats
             checkout/${{ matrix.app }}/Saved/MaterialStats
             checkout/${{ matrix.app }}/Saved/MaterialStatsDebug
+
+  marketplace-repack:
+    needs: [test]
+    name: Re-pack for UE Marketplace
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Note: these versions must match scripts/packaging/engine-versions.txt
+        unreal: ['5.1', '5.2', '5.3']
+
+    steps:
+      - name: Download packages
+        uses: actions/download-artifact@v3
+        with:
+          name: ${{ github.sha }}
+
+      - name: Extract packages
+        run: unzip sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip -d sentry-unreal-*-engine${{ matrix.unreal }}-marketplace
+
+      - name: Download Linux intermediates
+        uses: actions/download-artifact@v3
+        with:
+          name: Linux-intermediates-engine${{ matrix.unreal }}
+
+      - name: Extract Linux intermediates
+        run: unzip sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip -d sentry-unreal-*-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate
+
+      - name: Prepare Sentry packages for release
+        shell: pwsh
+        run: zip -r -1 -v sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip ./*
+
+      - name: Re-upload package
+        uses: actions/upload-artifact@v3.1.1
+        with:
+          # Artifact name is the commit sha. Which is what craft uses to find the relevant artifact.
+          name: ${{ github.sha }}
+          if-no-files-found: error
+          path: |
+            sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -242,7 +242,7 @@ jobs:
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/Intermediate/Build/Linux/*
+          path: /checkout/${{ matrix.app }}/Plugins/sentry/Intermediate/Build/Linux
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -289,11 +289,11 @@ jobs:
         uses: actions/download-artifact@v3
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: sentry-unreal-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate
+          path: LinuxIntermediate
 
       - name: Prepare Sentry packages for release
         shell: pwsh
-        run: zip -r -1 -v sentry-unreal-engine${{ matrix.unreal }}-marketplace.zip ./sentry-unreal-engine${{ matrix.unreal }}-marketplace/*
+        run: zip -ur sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip LinuxIntermediate
 
       - name: Re-upload package
         uses: actions/upload-artifact@v3.1.1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -281,10 +281,6 @@ jobs:
         with:
           name: ${{ github.sha }}
 
-      - name: Extract package
-        run:
-          unzip sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip -d sentry-unreal-engine${{ matrix.unreal }}-marketplace
-
       - name: Download Linux intermediates
         uses: actions/download-artifact@v3
         with:
@@ -302,4 +298,4 @@ jobs:
           name: ${{ github.sha }}
           if-no-files-found: error
           path: |
-            sentry-unreal-engine${{ matrix.unreal }}-marketplace.zip
+            sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -290,7 +290,7 @@ jobs:
           name: Linux-intermediates-engine${{ matrix.unreal }}
 
       - name: Extract Linux intermediates
-        run: unzip sentry-unreal-*-engine${{ matrix.unreal }}-marketplace.zip -d sentry-unreal-*-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate
+        run: unzip Linux-intermediates-engine${{ matrix.unreal }}.zip -d sentry-unreal-*-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate
 
       - name: Prepare Sentry packages for release
         shell: pwsh

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -237,7 +237,7 @@ jobs:
         id: pre-compile-linux
         run: |
           docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux
-          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al /dist
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al
 
       - name: Upload Linux intermediates
         uses: actions/upload-artifact@v3.1.1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -237,15 +237,13 @@ jobs:
         id: pre-compile-linux
         run: |
           docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux
-          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al
-          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al /Intermediate
-          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al /Intermediate/Build
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al dist
 
       - name: Upload Linux intermediates
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: checkout/${{ matrix.app }}/Plugins/sentry/Intermediate/Build/Linux
+          path: checkout/${{ matrix.app }}/Plugins/sentry/dist/Intermediate/Build/Linux/*
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -237,13 +237,13 @@ jobs:
         id: pre-compile-linux
         run: |
           docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux
-          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al dist
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al dist/Intermediate/Build/Linux
 
       - name: Upload Linux intermediates
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: checkout/${{ matrix.app }}/Plugins/sentry/dist/Intermediate/Build/Linux/*
+          path: checkout/${{ matrix.app }}/Plugins/sentry/dist/Intermediate/Build/Linux
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -237,6 +237,7 @@ jobs:
         id: pre-compile-linux
         run: |
           docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux -NoHostPlatform
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry ls -al /dist
 
       - name: Upload Linux intermediates
         uses: actions/upload-artifact@v3.1.1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -237,13 +237,12 @@ jobs:
         id: pre-compile-linux
         run: |
           docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux
-          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ls -al
 
       - name: Upload Linux intermediates
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/dist/Intermediate/Build/Linux          
+          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/Intermediate/Build/Linux          
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -242,7 +242,7 @@ jobs:
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/dist/HostProject/Intermediate/Build/Linux          
+          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/dist/HostProject/Plugins/Sentry/Intermediate/Build/Linux          
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -243,6 +243,7 @@ jobs:
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
+          if-no-files-found: error
           path: checkout/${{ matrix.app }}/Plugins/sentry/dist/Intermediate/Build/Linux
 
       - name: Build & package ${{ matrix.app }}
@@ -266,7 +267,7 @@ jobs:
 
   marketplace-repack:
     needs: [test]
-    name: Re-pack for UE Marketplace
+    name: Re-pack for marketplace UE ${{ matrix.unreal }}
     runs-on: ubuntu-latest
 
     strategy:
@@ -294,7 +295,6 @@ jobs:
       - name: Re-upload package
         uses: actions/upload-artifact@v3.1.1
         with:
-          # Artifact name is the commit sha. Which is what craft uses to find the relevant artifact.
           name: ${{ github.sha }}
           if-no-files-found: error
           path: |

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -288,9 +288,7 @@ jobs:
         uses: actions/download-artifact@v3
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-
-      - name: Extract Linux intermediates
-        run: unzip Linux-intermediates-engine${{ matrix.unreal }}.zip -d sentry-unreal-*-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate
+          path: sentry-unreal-*-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate
 
       - name: Prepare Sentry packages for release
         shell: pwsh

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -293,7 +293,7 @@ jobs:
 
       - name: Prepare Sentry packages for release
         shell: pwsh
-        run: zip -r -1 -v sentry-unreal-engine${{ matrix.unreal }}-marketplace.zip ./sentry-unreal-engine${{ matrix.unreal }}-marketplace/LinuxIntermediate/*
+        run: zip -r -1 -v sentry-unreal-engine${{ matrix.unreal }}-marketplace.zip ./sentry-unreal-engine${{ matrix.unreal }}-marketplace/*
 
       - name: Re-upload package
         uses: actions/upload-artifact@v3.1.1

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -233,6 +233,17 @@ jobs:
           path: |
             checkout/${{ matrix.app }}/Saved/Automation
 
+      - name: Pre-compile Linux intermediates
+        id: pre-compile-linux
+        run: |
+          docker exec -w /workspace/checkout/${{ matrix.app }}/Plugins/sentry unreal ue4 package -TargetPlatforms=Linux -NoHostPlatform
+
+      - name: Upload Linux intermediates
+        uses: actions/upload-artifact@v3.1.1
+        with:
+          name: Linux-intermediates-engine${{ matrix.unreal }}
+          path: /workspace/checkout/${{ matrix.app }}/Plugins/sentry/dist/Intermediate/Build/Linux          
+
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27
         if: ${{ matrix.unreal == '4.27'}}

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -242,7 +242,7 @@ jobs:
         uses: actions/upload-artifact@v3.1.1
         with:
           name: Linux-intermediates-engine${{ matrix.unreal }}
-          path: /checkout/${{ matrix.app }}/Plugins/sentry/Intermediate/Build/Linux
+          path: checkout/${{ matrix.app }}/Plugins/sentry/Intermediate/Build/Linux
 
       - name: Build & package ${{ matrix.app }}
         # Unreal 5.0 takes ~1 hour to build+package the app so only do it on v4.27

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,7 @@
 ### Fixes
 
 - Fix issue with invalidating breadcrumbs during event capturing on Win/Linux ([#445](https://github.com/getsentry/sentry-unreal/pull/445))
+- Fix build errors when cross-compiling for Linux on Windows with UE Marketplace plugin version ([#453](https://github.com/getsentry/sentry-unreal/pull/453))
 
 ### Dependencies
 

--- a/plugin-dev/Config/FilterPlugin.ini
+++ b/plugin-dev/Config/FilterPlugin.ini
@@ -7,7 +7,5 @@
 ;    /Extras/...
 ;    /Binaries/ThirdParty/*.dll
 /Scripts/*
-/Binaries/Linux/*
-/Binaries/Mac/*
-/Binaries/Win64/*
+/LinuxIntermediate/*
 /sentry-cli.properties

--- a/plugin-dev/Sentry.uplugin
+++ b/plugin-dev/Sentry.uplugin
@@ -37,6 +37,8 @@
 			"if [ -f \"$(PluginDir)/Scripts/upload-debug-symbols.sh\" ]; then\n sh \"$(PluginDir)/Scripts/upload-debug-symbols.sh\" $(TargetPlatform) $(TargetName) $(TargetType) $(TargetConfiguration) \"$(ProjectDir)\" \"$(PluginDir)\"\nelse\n echo \"Sentry: Symbol upload script is missing. Skipping post build step.\"\nfi"
 		],
 		"Win64": [
+			"set CRASHPAD_HANDLER=\"$(PluginDir)/Binaries/Linux/crashpad_handler\"",
+			"if \"$(TargetPlatform)\"==\"Linux\" (\n if not exist \"%CRASHPAD_HANDLER%\" (xcopy \"$(PluginDir)/Source/ThirdParty/Linux/bin/*\" \"$(PluginDir)/Binaries/Linux/\" /F /R /Y /I)\n)",
 			"set SYM_UPLOAD_SCRIPT=\"$(PluginDir)/Scripts/upload-debug-symbols-win.bat\"",
 			"if exist \"%SYM_UPLOAD_SCRIPT%\" (call \"%SYM_UPLOAD_SCRIPT%\" $(TargetPlatform) $(TargetName) $(TargetType) $(TargetConfiguration) \"$(ProjectDir)\" \"$(PluginDir)\") else (\necho Warning: Sentry: Symbol upload script is missing. Skipping post build step.\n)"
 		]

--- a/plugin-dev/Sentry.uplugin
+++ b/plugin-dev/Sentry.uplugin
@@ -28,6 +28,12 @@
 			"WhitelistPlatforms": [ "Win64", "Mac", "Android", "IOS", "Linux" ]
 		}
 	],
+	"PreBuildSteps":
+	{
+		"Win64": [
+			"if \"$(TargetPlatform)\"==\"Linux\" (\n if exist \"$(PluginDir)/LinuxIntermediate\" (\n if not exist \"$(PluginDir)/Intermediate/Build/Linux\" (echo | xcopy \"$(PluginDir)/LinuxIntermediate\" \"$(PluginDir)/Intermediate/Build/Linux\" /E /H /C /I) \n) \n)",
+		]
+	},
 	"PostBuildSteps":
 	{
 		"Mac": [

